crates/miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -331,16 +334,26 @@ export.get_block_hash
     padw push.BLK_HASH_PTR mem_loadw
 end
 
+#! Returns the block number of the reference block.
+#!
+#! Inputs:  []
+#! Outputs: [block_num]
+#!
+#! Where:
+#! - block_num is the number of the reference block for the transaction execution.
+export.get_block_num
```

Comment:
We should ideally call this `get_global_blk_num` (global to indicate it's the one from the public inputs and `blk` for consistency, although I would personally also like us to use "block", but that's another discussion).